is_iam_enabled.go

```diff
@@ -0,0 +1,76 @@
+package alks
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+)
+
+type IsIamRoleRequest struct {
+	RoleArn string `json:"roleArn"`
+}
+
+// IsIamRoleResponse is used to represent a role that's IAM active or not.
+type IsIamRoleResponse struct {
+	BaseResponse
+	RoleArn    string `json:"roleArn"`
+	IamEnabled bool   `json:"iamEnabled"`
+}
+
+// IsIamEnabled will check if a MI or STS assumed role is IAM active or not.
+func (c *Client) IsIamEnabled(roleArn string) (*IsIamRoleResponse, error) {
+	log.Printf("[INFO] Is IAM enabled for MI: %s", roleArn)
+
+	var iam IsIamRoleRequest
+	var body []byte
+	var err error
+
+	if len(roleArn) > 1 {
+		log.Printf("[INFO] Is IAM enabled for MI: %s", roleArn)
+
+		// Request for a MI.
+		iam = IsIamRoleRequest{
+			roleArn,
+		}
+
+		body, err = json.Marshal(struct {
+			IsIamRoleRequest
+		}{iam})
+	} else {
+		log.Printf("[INFO] Is IAM enabled for STS: %s", c.AccountDetails.Role)
+
+		// Request for STS
+		body, err = json.Marshal(struct {}{})
+	}
```

Comment:
You can simplify this logic in the request struct by providing the JSON struct tag `omitempty`.  That tells the runtime to not serialize that type if it's a `nil` or in this case, an empty string.